service/history/workflow/mutable_state_impl.go

```diff
@@ -5050,6 +5050,11 @@ func (ms *MutableStateImpl) GetUpdateCondition() (int64, int64) {
 func (ms *MutableStateImpl) SetSpeculativeWorkflowTaskTimeoutTask(
 	task *tasks.WorkflowTaskTimeoutTask,
 ) error {
+	taskID, err := ms.shard.GenerateTaskIDs(1)
+	if err != nil {
+		return err
+	}
+	task.TaskID = taskID[0]
```

Comment:
This should fix 0 `TaskID` for speculative WFT.

Reply:
this is for validateTaskGeneration?

Reply:
I just realized that missed it before and logs have it 0 everywhere. But you are right, it will help in `validateTaskGeneration` too because now this validation is just skipped.